packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -196,18 +203,15 @@ class OnsetPicker extends React.Component {
    * @param {ageUnit} - New ageUnit
    */
   changeAgeUnit(ageUnit) {
-    this.setState((prevState) => {
-      // Check if date can be calculated
-      const ageDate = Number.isInteger(prevState.age) && ageUnit
-        ? moment(this.props.birthdate).add(prevState.age, ageUnit) : undefined;
-      // Check if date is valid
-      const validDate = ageDate && ageDate >= moment(this.props.birthdate) && ageDate <= moment();
-
-      return {
-        ageUnit,
-        onsetDate: validDate ? ageDate : undefined,
-      };
-    }, this.handleOnsetUpdate);
+    const age = (this.state.age !== undefined) ? Number.isInteger(this.state.age) : undefined;
```

Comment:
`Number.isInteger` will return `false` on `undefined`, so I wouldn't bother with the check. Also wouldn't name this `age` as it's a boolean value, and not actually the age value. I'd also probably not set these values to `undefined`, when you're using them as booleans. Yes it's a falsey value, but why not just use `false`? Suggestion:

```javascript
const validAge = Number.isInteger(this.state.age);
// Attempt to calculate onset date
const ageDate = validAge && ageUnit ? moment(this.props.birthdate).add(this.state.age, ageUnit) : undefined;
// Check if date is valid
const validDate = validAge ? (ageDate && ageDate >= moment(this.props.birthdate) && ageDate <= moment()) : false;
```

Reply:
Resolved in 982e7aa